src/plugins/gui/providers/banxaProvider.ts

```diff
@@ -594,7 +594,54 @@ export const banxaProvider: FiatProviderFactory = {
             let insideInterval = false
 
             if (direction === 'buy') {
-              await showUi.openExternalWebView({ url: banxaQuote.data.order.checkout_url })
+              await showUi.openExternalWebView({
+                providerId,
+                url: banxaQuote.data.order.checkout_url,
+                deeplinkHandler: async link => {
+                  if (link.direction !== 'buy') return
+
+                  const orderResponse = await banxaFetch({ method: 'GET', url, hmacUser, path: `api/orders/${banxaQuote.data.order.id}`, apiKey })
+                  const order = asBanxaOrderResponse(orderResponse)
+
+                  switch (link.query.status) {
+                    case 'success': {
+                      console.log('Banxa WebView launch buy success: ' + link.uri)
+                      await showUi.trackConversion('Buy_Success', {
+                        conversionValues: {
+                          conversionType: 'buy',
+                          sourceFiatCurrencyCode: fiatCurrencyCode,
+                          sourceFiatAmount: priceQuote.fiat_amount,
+                          destAmount: new CryptoAmount({
+                            currencyConfig: coreWallet.currencyConfig,
+                            currencyCode: displayCurrencyCode,
+                            exchangeAmount: order.data.order.coin_amount
+                          }),
+                          fiatProviderId: providerId,
+                          orderId: banxaQuote.data.order.id
+                        }
+                      })
+                      await showUi.exitScene()
+                      break
+                    }
+                    case 'cancelled': {
+                      console.log('Banxa WebView launch buy cancelled: ' + link.uri)
```

Comment:
If you decided to correct the "canceled" string spelling, change it here too for searchability.